src/Messages.php

```diff
@@ -72,9 +72,16 @@ public function addMessage($key, $message)
         if (!isset($this->storage[$this->storageKey][$key])) {
             $this->storage[$this->storageKey][$key] = array();
         }
-        
+
         //Push onto the array
-        $this->storage[$this->storageKey][$key][] = (string)$message;
+        if (
+            is_array($message) === true
+            || is_object($message) === true
+        ) {
```

Comment:
I honestly would get rid of the checks and just have it append to the array without the (string)

//ping @silentworks @akrabat